2018-05-04  Hakan Selim Alpay  <hakan@lnxsrv06.seas.ucla.edu>

        * timespecNew.h (timespec_cmp): prevented integer overflow in timespec_cmp


--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ timespecNew.h	2018-05-04 18:27:17.024208000 -0700
@@ -45,8 +45,18 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  int ret;
+  if(a > b){
+    ret = 1;
+  }
+  else if(a == b){
+    ret = 0;
+  }
+  else{
+    ret = -1;
+  }
+  
+  return ret;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
